acpi.adoc

```diff
@@ -1,26 +1,142 @@
 [[acpi]]
 == ACPI Requirements
-=== ACPI Version
-=== ACPI-provided Data Structures
-=== ACPI Tables
-==== Mandatory ACPI Tables
-===== RSDP
-===== XSDT
-===== FADT
-===== DSDT and SSDT
-===== MADT
-===== RHCT
-===== SPCR
+
+The Advanced Configuration and Power Interface specification provides the OS-centric view of system configuration, various hardware resources, events and power management.
+
+This section defines mandatory and optional ACPI requirements on top of the ACPI specification cite:[ACPI]. Additional non-normative guidance may be found in <<ACPI Implementation Guidance>>.
+
+=== ACPI High-Level Requirements
+
+A compliant platform must:
+
+* Conform to UEFI specification requirements for ACPI:
+** RISC-V Platforms (cite:[UEFI] Section 2.3.7)
+** Industry Standard Configuration Tables (cite:[UEFI] Section 4.6.1.1)
+* Conform to version 6.6 of the ACPI specification, including RISC-V requirements listed in:
+** Multiple APIC Description Table (MADT, cite:[ACPI] Section 5.2.12)
+** RISC-V Hart Capabilities Table (RHCT, cite:[ACPI] Section 5.2.36)
+* Meet the requirements in <<Minimum Hardware Requirements Imposed By ACPI>>
+* Be 64-bits clean.
+** RSDT must not be implemented.
+** 32-bit address fields must be 0.
+* Implement a firmware-first APEI model (cite:[ACPI] Section 18.4)
+
+=== ACPI-specific Hardware Requirements
+
+This section lists the BRS-specific hardware requirements imposed by
+ACPI support.
+
+==== HW-Reduced ACPI Model
+
+Compliant RISC-V platforms only implement the HW-Reduced ACPI Model. This means the hardware portion of
+cite:[ACPI] (Section 4) is not required or supported. All functionality is instead provided through equivalent
+software-defined interfaces and the complexity in supporting ACPI is reduced.
+
+The following additional BRS requirements apply:
+
+* Both interrupt-signaled and GPIO-signaled ACPI Events are supported
```

Comment:
Do we need to call this out?

For gpio-signaled events that transitively indicates gpios are a requirement.

Reply:
It needs to be called out because the ACPI spec doesn't call out interrupt-signaled events as a GPE replacement choice. I do have an ASWG ticket for that, so once that is resolved we can remove the text.

Wrt gpios... not necessarily. HW-reduced ACPI supports both interrupt-signaled and GPIO-signaled ACPI events. So if you don't have GPIOS, you can use interrupt signaled (and even if you have GPIOs, you may choose to use interrupt signaled for simple enough use cases). We're providing choice so GPIO is not a requirement for BRS, it /may/ be a requirement depending on how you apply the ACPI spec. It doesn't hurt to call this out, but this might be in the guidance section instead.

Reply:
sg